sdk/core/Azure.Core.TestFramework/src/UseSyncMethodsInterceptor.cs

```diff
@@ -89,6 +89,13 @@ public void Intercept(IInvocation invocation)
                     Type[] modelType = returnType.GenericTypeArguments;
                     Type wrapperType = typeof(SyncPageableWrapper<>).MakeGenericType(modelType);
 
+                    Type resultType = result.GetType();
+
+                    // Check if wrapperType still contains generic paramters, requiring that we resolve the generic type arguments.
+                    if (wrapperType.ContainsGenericParameters && resultType.IsGenericType && resultType.GenericTypeArguments.Length > 0)
+                    {
+                        wrapperType = typeof(SyncPageableWrapper<>).MakeGenericType(resultType.GenericTypeArguments);
+                    }
```

Comment:
After unblocking the issue with `GetMethod` I ran into an issue where `TableClient.Query<T>`  was failing below at `Activator.CreateInstance` (error below) because `wrapperType.ContainsGenericParameters` == true. This was how I was able to resolve the underlying generic type. Any feedback welcome on if there is a better approach.

`Cannot create an instance of <foo> because Type.ContainsGenericParameters is true.`

Reply:
I think we can fix this by reading `returnType` of updated `methodInfo` after it was closed around the arguments.